mkosi/__init__.py

```diff
@@ -1259,14 +1259,32 @@ def gzip_binary(context: Context) -> str:
     return "pigz" if context.config.find_binary("pigz") else "gzip"
 
 
+def _kernel_get_ver(context: Context) -> str:
+    path = context.root / "usr/lib/modules"
+    # try to get version from the first dir under usr/lib/modules
+
+    if not path.exists():
+        return "unknown"
+
+    for item in path.iterdir():
```

Comment:
This could be combined into
```suggestion
    for item in path.glob("*"):
```
and if you relax the regex (assuming any dir under `/usr/lib/modules` starting with a digit is a kernel) it could just be
```suggestion
    for item in path.glob("[0-9]*"):
```

Reply:
Also, this does assume that the sorting order is reasonable and as you say, there will usually only be one, but people do weird stuff in scripts, so these assumptions might not hold generally, so it would be better to sort this by version (we have a version object for that).